JGI.py fixes

bkaras1 opened the pull request:
Plethora of fixes and changes to increase robustness, efficiency, and speed including:

- Updated CLI to argparse
- Fixed combined_taxon_id.json
- Removal of .dat temp files
- Reworked tracking of download process with creation and editing of organism url list json file
- Only queries JGI for organism list the first time script is run. Checks json and local files thereafter.
- Rewrote enzyme url creation
- Rewrote metadata and genome statistic dictionary creation

hbsmith commented on ecg/jgi.py:
```diff
@@ -109,75 +111,319 @@ def __get_domain_json(self,domain_url,database,domain):
-            full_url = self.homepage_url+html_suffix
+            
+            #full_url = self.homepage_url+html_suffix
+            full_url = "https://img.jgi.doe.gov/cgi-bin/m/main.cgi{}".format(html_suffix)
```
Can you change this to reference `self.homepage_url+"{}"`?

bkaras1 replied:
Sure.

hbsmith commented on ecg/jgi.py:
```diff
@@ -374,35 +672,41 @@ def _scrape_urls_unsafe(self, path, domain, organism_urls,
+                pbar.set_description("Scraping %s ..."%(re.search(r'\d+$', organism_url).group(0)))
+                taxon_id, org_dict = self.__scrape_organism_url_from_metagenome_domain(domain_path,organism_url,assembly_types)
+
+                organism_urls[organism_url] = 0
```
Can't the pruning function be avoided entirely if the organism_url keys are just deleted here instead of set to 0? Is there any reason not to do that?

bkaras1 replied:
I have the pruning function to check if there are existing files present in the directory and to skip them. I set the organism_url keys to 0 in case there is an error in the last one downloaded and you can just reset the json file to 1 for that organism url by manually editing the file. Also, it lets me test and retest batches of urls by just deleting the json files and resetting their keys to 1 without having to do a large query that is prone to timeouts like Bacteria or Metagenomes.

hbsmith commented on ecg/jgi.py:
```diff
@@ -374,35 +672,41 @@ def _scrape_urls_unsafe(self, path, domain, organism_urls,
+                pbar.set_description("Scraping %s ..."%(re.search(r'\d+$', organism_url).group(0)))
+                taxon_id, org_dict = self.__scrape_organism_url_from_regular_domain(domain_path,organism_url)
+
+                organism_urls[organism_url] = 0
```
same comment as above

bkaras1 commented:
Updated documentation/examples for CLI for ecg.py, jgi.py, and kegg.py. Updated CLI to argparse for ecg.py, jgi.py, and kegg.py. Made minor fixes to CLI for jgi.py and kegg.py with respect to passing lists as arguments.
